packages/m2m-gateway/src/utils/validators/purposeValidator.ts

```diff
@@ -0,0 +1,11 @@
+import { purposeApi } from "pagopa-interop-api-clients";
+import { missingActivePurposeVersion } from "../../model/errors.js";
+
+export function assertPurposeVersionExists(
+  purposeVersion: purposeApi.PurposeVersion | undefined,
+  purposeId: purposeApi.Purpose["id"]
+): asserts purposeVersion is NonNullable<purposeApi.PurposeVersion> {
+  if (!purposeVersion) {
+    throw missingActivePurposeVersion(purposeId);
+  }
+}
```

Comment:
This seems wrong by reading it: the assertion checks that a purpose version exists, without any mention to the purpose version state - but the error is `missingActivePurposeVersion` and also the message of the error says "There is no active version for purpose". We call this when looking for a Draft version so the two things don't match...
See also the comment above on the error

```suggestion
export function assertPurposeVersionExistsWithState(
  purposeVersion: purposeApi.PurposeVersion | undefined,
  purposeId: purposeApi.Purpose["id"],
  state: purposeApi.PurposeVersionState
): asserts purposeVersion is NonNullable<purposeApi.PurposeVersion> {
  if (!purposeVersion) {
    throw missingActivePurposeVersionWithState(purposeId, state);
  }
}
```